<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Lab1</class>
 <widget class="QDialog" name="Lab1">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>486</width>
    <height>629</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Information defense</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Source data</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <item row="0" column="1">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Message</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1" rowspan="2" colspan="2">
       <widget class="QPlainTextEdit" name="editMessage">
        <property name="plainText">
         <string>ГРУЗИТЕАПЕЛЬСИНЫ</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="lblRequiredLength">
        <property name="text">
         <string>Required length: Unknown</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLabel" name="labelInputLength">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Text length: 0</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QCheckBox" name="checkRemoveSpaces">
        <property name="text">
         <string>Delete spaces</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Encryption keys</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="2" column="0" colspan="3">
       <widget class="QLineEdit" name="editFirstKey">
        <property name="text">
         <string>4132</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="3">
       <widget class="QLineEdit" name="editSecondKey">
        <property name="text">
         <string>3142</string>
        </property>
       </widget>
      </item>
      <item row="0" column="2">
       <widget class="QLabel" name="lblKey1Length">
        <property name="text">
         <string>Length: 0</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Second key</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2">
       <widget class="QLabel" name="lblKey2Length">
        <property name="text">
         <string>Length: 0</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>First key</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Output data</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="1" column="1">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Result</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QPlainTextEdit" name="editResult"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>Actions</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="btnEncrypt">
        <property name="text">
         <string>Encode</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="btnDecrypt">
        <property name="text">
         <string> Decode</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
